Exercícios do bloco 21.1 da Trybe

rosids opened the pull request:
#### PR dos exercícios do bloco 21.1

- [x] 01. Escreva uma query que exiba o maior salário da tabela.
- [x] 02. Escreva uma query que exiba a diferença entre o maior e o menor salário.
- [x] 03. Escreva uma query que exiba a média salarial de cada `JOB_ID`, ordenando pela média salarial em ordem decrescente.
- [x] 04. Escreva uma query que exiba a quantidade de dinheiro necessária para realizar o pagamento de todas as pessoas funcionárias.
- [x] 05. Escreva uma query que exiba quatro informações: o maior salário, o menor salário, a soma de todos os salários e a média dos salários. Todos os valores devem ser formatados para ter apenas duas casas decimais.
- [x] 06. Escreva uma query que exiba a quantidade de pessoas que trabalham como pessoas programadoras (`IT_PROG`).
- [x] 07. Escreva uma query que exiba a quantidade de dinheiro necessária para efetuar o pagamento de cada profissão (`JOB_ID`).
- [x] 08. Utilizando a query anterior, faça as alterações para que seja exibido somente a quantidade de dinheiro necessária para cobrir a folha de pagamento das pessoas programadoras (`IT_PROG`).
- [x] 09. Escreva uma query que exiba em ordem decrescente a média salarial de todos os cargos, exceto das pessoas programadoras (`IT_PROG`).
- [x] 10. Escreva um query que exiba média salarial e o número de funcionários de todos os departamentos com mais de dez funcionários. **Dica:** agrupe pelo `department_id`.
- [x] 11. Escreva uma query que atualize a coluna `PHONE_NUMBER`, de modo que todos os telefones iniciados por 515 agora devem iniciar com `777`.
- [x] 12. Escreva uma query que só exiba as informações dos funcionários cujo o primeiro nome tenha oito ou mais caracteres.
- [x] 13. Escreva uma query que exiba as seguintes informações de cada funcionário: `id`, `primeiro nome` e `ano no qual foi contratado` (exiba somente o ano).
- [x] 14. Escreva uma query que exiba as seguintes informações de cada funcionário: `id`, `primeiro nome` e `dia do mês no qual foi contratado` (exiba somente o dia).
- [x] 15. Escreva uma query que exiba as seguintes informações de cada funcionário: `id`, `primeiro nome` e `mês no qual foi contratado` (exiba somente o mês).
- [x] 16. Escreva uma query que exiba os nomes dos funcionários em letra maiúscula.
- [x] 17. Escreva uma query que exiba o sobrenome e a data de contratação de todos os funcionário contratados em julho de 1987.
- [x] 18. Escreva uma query que exiba as seguintes informações de cada funcionário: `nome`, `sobrenome`, `tempo que trabalha na empresa (em dias)`.

rosids committed:
docs: add readme with instructions from exercises 21.1

rosids committed:
feat: add the hr database for exercises 21.1

rosids committed:
feat: add exercise resolutions from block 21.1

rosids committed:
docs: block 21.1 - "Most used functions in SQL" completed

paulohbsimoes commented on desenvolvimento-back-end/bloco_21/dia_01/exercises.sql:
```diff
@@ -0,0 +1,54 @@
+
+# Exercício 11
+SET SQL_SAFE_UPDATES = 0;
+UPDATE hr.employees SET PHONE_NUMBER = REPLACE(PHONE_NUMBER, '515', '777') WHERE PHONE_NUMBER LIKE '515%';
```
Um ponto de atenção para essa query é que se houver algum número que tenha 515 em alguma outra parte além do início, essa parte também será alterada. Por exemplo, '515-888-515' vira '777-888-777'.

rosids replied:
> Um ponto de atenção para essa query é que se houver algum número que tenha 515 em alguma outra parte além do início, essa parte também será alterada. Por exemplo, '515-888-515' vira '777-888-777'.

@paulohbsimoes, obrigada pelo alerta! Preciso utilizar o substring né?

paulohbsimoes reviewed (commented):
Boa Rosii! :rocket: :smile: 

![good_job](https://user-images.githubusercontent.com/66493294/115158884-d2486400-a066-11eb-956d-b4998f81c375.gif)

rosids committed:
feat: Correction of exercise 11 of block 21.1
 - Fixes changing phone numbers only when it starts with "515", without changing the rest of the phone numbers
